Basic Components (Milestone 2)

mosesogwo opened the pull request:
This pull request covers the setup of the following:
- App component
- Display component
- ButtonPanel component and
- Button Component.

mosesogwo committed:
create component folder, four components files

mosesogwo committed:
setup Display component to receive result as props from app

mosesogwo committed:
add buttons to ButtonPanel component with name passed as props

mosesogwo committed:
export each component fromm each file, replace div in button comp with button tag

mosesogwo committed:
change div in button component to button tag

mosesogwo committed:
add propType validations for Display and Button components

mosesogwo committed:
fix eslint use of destructuring requirement to pass props in App component

bolah2009 commented on src/components/ButtonPanel.js:
```diff
@@ -0,0 +1,42 @@
+    <div className="group-2">
+      <Button name="7" />
+      <Button name="8" />
+      <Button name="9" />
```
[Optional]

We could rewrite the ButtonPanel component to use two loops to make the `Button` instead of hardcoding them. By doing this, we keep our code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

We could achieve this by creating an array of symbol and loop through the array using [`map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) to create the ButtonPanel component.

bolah2009 commented on src/components/Display.js:
```diff
@@ -0,0 +1,12 @@
+
+Display.propTypes = {
+  result: PropTypes.string.isRequired,
+};
```
Let's give `Display` component a default prop of string `0`. See https://reactjs.org/docs/typechecking-with-proptypes.html#default-prop-values

> its result prop should be a String with defualt value = 0 ([should be String screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/validateAsString.png)) ([default props screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/DefaultProp.png))

bolah2009 reviewed (changes requested):
Hi @mosesogwo 

Only one required changes, please see the comment below for more details.

#### :recycle:  Make these changes and submit another code review request. 👍🏾

#### Feel free to reach me on slack if you having issues implementing these features. :relaxed: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

mosesogwo committed:
add defaultProps value for result, value is string 0

mosesogwo committed:
modify Display component to loop through array of button symbols instead of hardcoding

mosesogwo committed:
fix unique key for siblings in loop with crypto-js package

mosesogwo commented:
Thanks for your review.

- I have added the defaultProps for result in DIsplay component as "0".
- I have modified ButtonPanel to use `Array map` instead of hardcoding buttons.

JamesOkunlade reviewed (approved):
# Approved 😁 ✅ 

Hi @mosesogwo , great work implementing the changes requested by the previous reviewer 🙂 

This `milestone2/feature-branch` now meets the minimum requirements for approval. Merge to `development` and move on to the `next-milestone`

**Goodluck** 😁 
[James](https://twitter.com/JamesOkunlade)
